mpolyakov/homework_4/olx_scraping/olx_scraping/spiders/product_spider.py

```diff
@@ -0,0 +1,68 @@
+import scrapy
+from scrapy.exceptions import CloseSpider
+
+from olx_scraping.items import Product
+
+
+class ProductSpider(scrapy.Spider):
+    name = 'products'
+
+    custom_settings = {
+        'FEED_URI': 'output.json',
+        'FEED_EXPORT_ENCODING': 'utf-8',
+        'CLOSESPIDER_ITEMCOUNT': 10,
+    }
+
+    allowed_domains = ['olx.ua']
+    start_urls = ['https://www.olx.ua/elektronika/']
+
+    MAX_COUNT = 10  # defines maximum items to store
+    count = 0
+
+    def parse(self, response):
+        """
+        Find links to product page.
+        """
+        product_page_links = response.css('.detailsLink')
+        yield from response.follow_all(product_page_links, self.parse_product)
+
+        pagination_links = response.css('span.fleft a')
+        yield from response.follow_all(pagination_links, self.parse)
```

Comment:
nice :+1:

Reply:
btw, do you know the meaning of the yield from?

Reply:
It's the same as:
`for a in response.css('span.fleft a'):
    yield response.follow(a, self.parse_product)`